package/upgrade/upgrade_manifests.sh

```diff
@@ -1256,6 +1258,23 @@ patch_local_cluster_details() {
   kubectl patch -n fleet-local cluster.provisioning local --subresource=status --type=merge --patch '{"status":{"fleetWorkspaceName": "fleet-local"}}'
 }
 
+# RedeployAgentGeneration can be used to force redeploying the agent.
+# RedeployAgentGeneration int64 `json:"redeployAgentGeneration,omitempty"`
+patch_fleet_cluster() {
+  local generation=$(kubectl get -n fleet-local cluster.fleet local -o jsonpath='{.status.agentDeployedGeneration}')
+  local new_generation=$((generation+1))
+  patch_manifest="$(mktemp --suffix=.json)"
+  cat > "$patch_manifest" <<EOF
+{
+  "spec": {
+    "redeployAgentGeneration": $new_generation
+  }
+}
+EOF
+  echo $patch_string |  kubectl patch -n fleet-local cluster.fleet local  --type=merge --patch-file $patch_manifest
```

Comment:
Maybe we could drop the `echo $patch_string` part?

Reply:
`kubectl patch clusters.fleet.cattle.io -n fleet-local local --type=json -p '[{"op": "add", "path": "/spec/redeployAgentGeneration", "value": '"$new_generation"'}]'` works without a file, but it reads a bit more cryptic.

Reply:
i have no strong opinions about it, other than to force a redeploy of the agent 😆